PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+     */
+    public let periodType: PeriodType
+
+    /**
+     The latest purchase or renewal date for the entitlement.
+     */
+    public let latestPurchaseDate: Date? // TODO: This used to be NON-NULL
+
+    /**
+     The first date this entitlement was purchased
+     */
+    public let originalPurchaseDate: Date? // TODO: This used to be NON-NULL
+
+    /**
+     The expiration date for the entitlement, can be `nil` for lifetime access.
+     If the `periodType` is `trial`, this is the trial expiration date.
+     */
+    public private(set) var expirationDate: Date?
+
+    /**
+     The store where this entitlement was unlocked from
+     Either: RCAppStore, RCMacAppStore, RCPlayStore, RCStripe, RCPromotional, RCUnknownStore
+     */
+    public let store: Store
+
+    /**
+     The product identifier that unlocked this entitlement
+     */
+    public let productIdentifier: String
+
+    /**
+     False if this entitlement is unlocked via a production purchase
+     */
+    public let isSandbox: Bool
+
+    /**
+     The date an unsubscribe was detected. Can be `nil`.
+
+     Note: Entitlement may still be active even if user has unsubscribed. Check the `isActive` property.
+     */
+    public private(set) var unsubscribeDetectedAt: Date?
+
+    /**
+     The date a billing issue was detected. Can be `nil` if there is no
+     billing issue or an issue has been resolved.
+
+     Note: Entitlement may still be active even if there is a billing issue.
+     Check the `isActive` property.
+     */
+    public private(set) var billingIssueDetectedAt: Date?
+
+    /**
+     Use this property to determine whether a purchase was made by the current user
+     or shared to them by a family member. This can be useful for onboarding users who have had
+     an entitlement shared with them, but might not be entirely aware of the benefits they now have.
+     */
+    public let ownershipType: PurchaseOwnershipType
+
+    // TODO(post-migration): Make this internal
+    // TODO(cleanup): Codable
+    public init(entitlementId: String,
+                entitlementData: [String: Any],
+                productData: [String: Any],
+                dateFormatter: DateFormatter,
+                requestDate: Date?) {
+        // Entitlement data
+        let entitlementExpiresDateString = entitlementData["expires_date"] as? String
+        let entitlementPurchaseDateString = entitlementData["purchase_date"] as? String
+        let productIdString = entitlementData["product_identifier"] as? String
+
+        // Product data
+        let periodTypeString = productData["period_type"] as? String
+        let originalPurchaseDateString = productData["original_purchase_date"] as? String
+        let productExpiresDateString = productData["expires_date"] as? String
+        let storeString = productData["store"] as? String
+        let maybeSandbox = productData["is_sandbox"] as? NSObject // This could be a String or NSNumber
```

Comment:
preeetty we can trace it back to find out its actual type. This is a value that comes from [our REST API](https://docs.revenuecat.com/reference#subscribers). 
So if you debug the value as it comes in you can see whether it gets translated into `String` or `NSNumber` (I don't recall which one it is tbh)

Reply:
✅